proxy_multi_map.go

```diff
@@ -0,0 +1,515 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package hazelcast
+
+import (
+	"context"
+	"time"
+
+	"github.com/hazelcast/hazelcast-go-client/internal/proto"
+	"github.com/hazelcast/hazelcast-go-client/internal/proto/codec"
+	"github.com/hazelcast/hazelcast-go-client/internal/serialization"
+	"github.com/hazelcast/hazelcast-go-client/types"
+)
+
+const (
+//	maxIndexAttributes               = 255
+//	defaultLockID                    = 0
+//	lockIDKey          lockIDKeyType = "__hz_lock_id"
+//	leaseUnset                       = -1
+)
+
+//type lockID int64
+//type lockIDKeyType string
+
+/*
+MultiMap is a distributed map.
+Hazelcast Go client enables you to perform operations like reading and writing from/to a Hazelcast MultiMap with methods like Get and Put.
+For details, see https://docs.hazelcast.com/hazelcast/latest/data-structures/multimap.html
+
+Listening for MultiMap Entry Events
+
+The first step of listening to entry-based events is  creating an instance of MultiMapEntryListenerConfig.
+MultiMapEntryListenerConfig contains options to filter the events by key and has an option to include the value of the entry, not just the key.
+You should also choose which type of events you want to receive.
+In the example below, a listener configuration for added and updated entries is created.
+Entries only with key "somekey":
+
+	entryListenerConfig := hazelcast.MultiMapEntryListenerConfig{
+		Key: "somekey",
+		IncludeValue: true,
+	}
+	m, err := client.GetMap(ctx, "somemap")
+	entryListenerConfig.NotifyEntryAdded(true)
+	entryListenerConfig.NotifyEntryUpdated(true)
```

Comment:
Should `entryListenerConfig` lines moved above the `GetMap` line?

Reply:
That makes sense, thanks